packages/website-backend/test/unit/api/controllers/RealTimeReportsController.spec.ts

```diff
@@ -0,0 +1,366 @@
+      const stub = sinon.createStubInstance(MutationEventResponseHandler);
+      responseHandlerForProjectStub.returns(stub);
+      findReportStub.resolves({
+        files: {},
+        schemaVersion: '1',
+        thresholds: { high: 80, low: 60 },
+        moduleName: 'core',
+        projectName: 'github.com/user/does-not-exist/',
+        version: 'master',
+        mutationScore: 89,
+      });
+      getEventsStub.returns(
+        Promise.resolve([
+          { id: '1', status: MutantStatus.Killed },
+          { id: '2', status: MutantStatus.Survived },
+        ])
+      );
+
+      await request.get('/api/real-time/github.com/user/does-not-exist/master');
+
+      expect(getEventsStub).calledWith({
+        projectName: 'github.com/user/does-not-exist',
+        version: 'master',
+        moduleName: undefined,
+        realTime: true,
+      });
+      expect(stub.sendMutantTested.calledTwice).to.be.true;
+    });
+  });
+
+  describe('HTTP POST /*', () => {
+    let serverStub: sinon.SinonStubbedInstance<MutationEventResponseHandler>;
+
+    beforeEach(() => {
+      serverStub = sinon.createStubInstance(MutationEventResponseHandler);
+      responseHandlerForProjectStub.returns(serverStub);
+    });
+
+    it('should return unauthorized if the `X-Api-key` header is not set', async () => {
+      const response = await request.post(
+        '/api/real-time/github.com/user/does-exist/master'
+      );
+
+      expect(response.status).to.eq(401);
+    });
+
+    it('shoud not send any event if no mutant is sent', async () => {
+      await request
+        .post('/api/real-time/github.com/user/does-not-exist/master')
+        .set('X-Api-Key', apiKey)
+        .send([])
+        .expect(200);
+
+      expect(responseHandlerForProjectStub.calledOnce).to.be.true;
+      expect(serverStub.sendMutantTested.notCalled).to.be.true;
+    });
+
+    it('should return a bad request if null is sent', async () => {
+      await request
+        .post('/api/real-time/github.com/user/does-not-exist/master')
+        .set('X-Api-Key', apiKey)
+        .send(null as unknown as undefined) // trust me
+        .expect(400);
+    });
+
+    it('should return a bad request if undefined is sent', async () => {
+      await request
+        .post('/api/real-time/github.com/user/does-not-exist/master')
+        .set('X-Api-Key', apiKey)
+        .send(undefined)
+        .expect(400);
+    });
+
+    it('should send a mutant tested for every client connected', async () => {
+      await request
+        .post('/api/real-time/github.com/user/does-not-exist/master')
+        .set('X-Api-Key', apiKey)
+        .send([{ id: '1', status: MutantStatus.Killed }])
+        .expect(200);
+
+      expect(responseHandlerForProjectStub.calledOnce).to.be.true;
+      expect(serverStub.sendMutantTested.firstCall.firstArg).to.deep.include({
+        id: '1',
+        status: MutantStatus.Killed,
+      });
+    });
```

Comment:
Missing a test here: `'should not send a mutant tested to a disconnected client.'`.

Reply:
I don't think that's possible to do since we don't have an actual instance of the `MutationEventResponseOrchestrator`. The name of the test above this one is incorrect, will change that.